test/success/conflict.prl

```diff
@@ -0,0 +1,12 @@
+// The term syntax `f(foo)` is ambiguous: is it a custom operator f, or is
+// it a variable f with foo surrounded by parens.
+Thm custom-op-versus-parens : [ (f : (x : bool) -> bool) -> bool true ] by [
+  'lam([f]. f (f tt))
```

Comment:
Ahh, this is a very good question, and it's not quite clear what to do here. I think that in the end, we will want to have a stage of parsing that comes even before AST, at which we have some sort of ambiguous "identifier applied to a spine of terms" constructor, which the elaborator interprets in the appropriate way based on the information it has about the current signature and local context—choosing either a custom operator application, a metavariable application or a function application. I don't think it's crucial to fix this right now, but we should keep it on the radar.